src/wp-includes/block-template-utils.php

```diff
@@ -669,11 +668,10 @@ function _wp_build_title_and_description_for_taxonomy_block_template( $taxonomy,
 		'number' => 2,
 		'name'   => $term_title,
 	);
-	$args = wp_parse_args( $args, $default_args );
-
-	$terms_with_same_title = $term_query->query( $args );
+	$args                        = wp_parse_args( $args, $default_args );
+	$terms_with_same_title_query = $term_query->query( $args );
 
-	if ( count( $terms_with_same_title ) > 1 ) {
+	if ( count( $terms_with_same_title_query->terms ) > 1 ) {
```

Comment:
This should not be needed. `query` returns the array of terms. But it can't hurt.